lisa/tools/virtualclient.py

```diff
@@ -0,0 +1,255 @@
+            f"{self.command} {cmd}", shell=True, sudo=True
+        )
+        return process
+
+    def _build_command(
+        self,
+        profile_name: str,
+        client_id: str,
+        port: int,
+        clean_target: str,
+        content_path: str,
+        debug: bool,
+        event_hub_connection_str: str,
+        system: str,
+        flush_wait: int,
+        experiment_id: str,
+        fail_fast: bool,
+        dependencies: bool,
+        iterations: int,
+        layout_path: str,
+        log_retention: int,
+        log_to_file: bool,
+        log_level: int,
+        meta_data: str,
+        packages: str,
+        parameters: str,
+        proxy_api: str,
+        seed: int,
+        scenarios: str,
+        timeout: int,
+    ) -> str:
+        cmd_parts = [
+            f" --profile={self._vc_profile_file_path}/{profile_name}.json ",
+            f" --timeout={timeout}",
+        ]
+
+        def add_if_non_empty(option: str, value: Any) -> None:
+            if value:
+                cmd_parts.append(f" {option}={value} ")
+
+        def add_if_true(option: str, flag: Any) -> None:
+            if flag:
+                cmd_parts.append(f" {option} ")
+
+        add_if_non_empty("--clientId", client_id)
+        add_if_non_empty("--port", port)
+        add_if_non_empty("--clean", clean_target)
+        add_if_non_empty("--cp", content_path)
+        add_if_true("--verbose", debug)
+        add_if_non_empty("--eventHubConnectionString", event_hub_connection_str)
+        add_if_non_empty("--system", system)
+        add_if_non_empty("--flush-wait", flush_wait)
+        add_if_non_empty("--experimentId", experiment_id)
+        add_if_true("--fail-fast", fail_fast)
+        add_if_true("--dependencies", dependencies)
+        add_if_non_empty("--iterations", iterations)
+        if layout_path:
+            cmd_parts.append(f" --layoutPath={self.node.get_pure_path(layout_path)} ")
+        add_if_non_empty("--log-retention", log_retention)
+        add_if_true("--log-to-file", log_to_file)
+        add_if_non_empty("--log-level", log_level)
+        add_if_non_empty("--metadata", meta_data)
+        add_if_non_empty("--packages", packages)
+        add_if_non_empty("--parameters", parameters)
+        add_if_non_empty("--proxy-api", proxy_api)
+        add_if_non_empty("--seed", seed)
+        add_if_non_empty("--scenarios", scenarios)
+
+        return "".join(cmd_parts)
+
+    def download_raw_data_file(self) -> Path:
+        local_path = self.node.local_log_path / "metrics.csv"
+        remote_path = self.node.get_pure_path(f"{self._vc_log_path}/metrics.csv")
+        current_user = self.node.tools[Whoami].get_username()
+        self.node.tools[Chown].change_owner(remote_path, current_user)
+        self.node.shell.copy_back(
+            remote_path,
+            local_path,
+        )
+        return local_path
+
+    def create_metrics_message(self, metrics_file: Path) -> None:
```

Comment:
The method name should be like `send_metrics_message`, because it sends message, not only create.